aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -182,15 +186,31 @@ static void maybe_promote_common_dtype(OperandInfo& op, ScalarType common_dtype)
 
 void TensorIterator::compute_types() {
   bool missing_dtypes = false;
+  bool missing_output_dtypes = false;
```

Comment:
What is the utility of providing both `missing_output_dtypes` and `has_read_write_op` as arguments?

Reply:
I assume that we can compute and promote stype based only on inputs, only if there is no missing output dtypes and there is no outputs that are same as inputs(inplace)